File: demos/javascript/dsa/linear/dynamic_array.js
console.log("=== JavaScript Array & Deque Demo ===");
const arr = [10, 20];
arr.push(30);
console.assert(arr.length === 3, "Length assertion failed");
console.assert(arr[1] === 20, "Index assertion failed");
console.assert(arr.pop() === 30, "Pop assertion failed");

const deque = ["center"];
deque.unshift("front");
deque.push("back");
console.assert(deque[0] === "front" && deque[2] === "back", "Deque assertion failed");

console.log("JS Array:", arr, "Deque:", deque);
console.log("JavaScript Dynamic Array tests passed successfully.");

File: demos/ruby/dsa/linear/dynamic_array.rb
puts "=== Ruby Array & Deque Demo ==="
arr = [10, 20]
arr.push(30)
raise "Size error" unless arr.length == 3
raise "Index error" unless arr[1] == 20
raise "Pop error" unless arr.pop == 30

deque = ["center"]
deque.unshift("front")
deque.push("back")
raise "Deque error" unless deque.first == "front" && deque.last == "back"

puts "Ruby Array: #{arr}, Deque: #{deque}"
puts "Ruby Dynamic Array tests passed successfully."
